src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -0,0 +1,2458 @@
+        log += (zoneIndex + 1);
+        addLog(LOG_LEVEL_INFO, log);
+      }
+      #  endif // ifdef P104_DEBUG_DEV
+    }
+    # endif    // ifdef P104_USE_ZONE_ACTIONS
+    zoneIndex = zCounter + zoneOffset;
+
+    if (action == P104_ACTION_DELETE) {
+      zoneOffset--;
+    } else {
+      # ifdef P104_DEBUG_DEV
+
+      if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+        log  = F("P104: read zone: ");
+        log += (zoneIndex + 1);
+        addLog(LOG_LEVEL_INFO, log);
+      }
+      # endif // ifdef P104_DEBUG_DEV
+      zones.push_back(P104_zone_struct(zoneIndex + 1));
+
+      zones[zoneIndex].size          = getFormItemIntCustomArgName(index + P104_OFFSET_SIZE);
+      zones[zoneIndex].text          = enquoteString(webArg(getPluginCustomArgName(index + P104_OFFSET_TEXT)));
+      zones[zoneIndex].content       = getFormItemIntCustomArgName(index + P104_OFFSET_CONTENT);
+      zones[zoneIndex].alignment     = getFormItemIntCustomArgName(index + P104_OFFSET_ALIGNMENT);
+      zones[zoneIndex].animationIn   = getFormItemIntCustomArgName(index + P104_OFFSET_ANIM_IN);
+      zones[zoneIndex].speed         = getFormItemIntCustomArgName(index + P104_OFFSET_SPEED);
+      zones[zoneIndex].animationOut  = getFormItemIntCustomArgName(index + P104_OFFSET_ANIM_OUT);
+      zones[zoneIndex].pause         = getFormItemIntCustomArgName(index + P104_OFFSET_PAUSE);
+      zones[zoneIndex].font          = getFormItemIntCustomArgName(index + P104_OFFSET_FONT);
+      zones[zoneIndex].layout        = getFormItemIntCustomArgName(index + P104_OFFSET_LAYOUT);
+      zones[zoneIndex].specialEffect = getFormItemIntCustomArgName(index + P104_OFFSET_SPEC_EFFECT);
+      zones[zoneIndex].offset        = getFormItemIntCustomArgName(index + P104_OFFSET_OFFSET);
+
+      if (zones[zoneIndex].size != 0) { // for newly added zone, use defaults
+        zones[zoneIndex].brightness  = getFormItemIntCustomArgName(index + P104_OFFSET_BRIGHTNESS);
+        zones[zoneIndex].repeatDelay = getFormItemIntCustomArgName(index + P104_OFFSET_REPEATDELAY);
+      }
+    }
+    # ifdef P104_DEBUG_DEV
+
+    if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+      log  = F("P104: add zone: ");
+      log += (zoneIndex + 1);
+      addLog(LOG_LEVEL_INFO, log);
+    }
+    # endif // ifdef P104_DEBUG_DEV
+
+    # ifdef P104_USE_ZONE_ACTIONS
+
+    if (((action == P104_ACTION_ADD_BELOW) && (zoneOrder == 0)) ||
+        ((action == P104_ACTION_ADD_ABOVE) && (zoneOrder == 1))) {
+      zones.push_back(P104_zone_struct(0));
+      zoneOffset++;
+      #  ifdef P104_DEBUG_DEV
+
+      if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+        log  = F("P104: insert after zone: ");
+        log += (zoneIndex + 2);
+        addLog(LOG_LEVEL_INFO, log);
+      }
+      #  endif // ifdef P104_DEBUG_DEV
+    }
+    # endif    // ifdef P104_USE_ZONE_ACTIONS
+
+    index += P104_OFFSET_COUNT;
+    delay(0);
+  }
+
+  uint16_t bufferSize;
+  char    *settingsBuffer;
```

Comment:
This `settingsBuffer` is not needed, since you're copying it later from a `String` object.
So you can also use that `String` as buffer to save.

Reply:
Removed it.